skpro/regression/bayesian.py

```diff
@@ -0,0 +1,391 @@
+        # if writes derived values to self, should *not* overwrite self.parama etc
+        # instead, write to self._parama, self._newparam (starting with _)
+
+        # todo: default estimators should have None arg defaults
+        #  and be initialized here
+        #  do this only with default estimators, not with parameters
+        # if est2 is None:
+        #     self.estimator = MyDefaultEstimator()
+
+        # todo: if tags of estimator depend on component tags, set these here
+        #  only needed if estimator is a composite
+        #  tags set in the constructor apply to the object and override the class
+        #
+        # example 1: conditional setting of a tag
+        # if est.foo == 42:
+        #   self.set_tags(handles-missing-data=True)
+        # example 2: cloning tags from component
+        #   self.clone_tags(est2, ["enforce_index_type", "handles-missing-data"])
+
+    # todo: implement this, mandatory
+    def _fit(self, X, y):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            feature instances to fit regressor to
+        y : pandas DataFrame, must be same length as X
+            labels to fit regressor to
+
+        Returns
+        -------
+        self : reference to self
+        """
+        # insert logic for estimator here
+        # fitted parameters should be written to parameters ending in underscore
+
+        # self must be returned at the end
+        assert isinstance(X, pd.DataFrame), "X must be a pd.DataFrame!"
+        assert isinstance(y, pd.DataFrame), "y must be a pd.DataFrame!"
+        assert len(y.columns) == 1, "y must have only one column!"
+        self.X = X
+        self.y = y
+        self.y_vals = y.values[:,0] # we need a 1-dimensional array for compatibility with pymc 
+        self.X_cols = X.columns
+        self.y_cols = y.columns
+
+        with pm.Model() as self.model:
+            # Mutable data containers
+            X_data = pm.MutableData("X", self.X, dims = ("obs_id", "pred_id"))
+            y_data = pm.MutableData("y", self.y_vals, dims = ("obs_id"))
+
+            # Priors for unknown model parameters
+            self.intercepts = pm.Normal("intercepts", mu=0, sigma=1)
+            self.slopes = pm.Normal("slopes", mu=0, sigma=1, dims=("pred_id"))
+            self.sigma = pm.HalfNormal("sigma", sigma=1)
+
+            # Expected value of outcome
+            self.mu = pm.Deterministic("mu", self.intercepts + pm.math.dot(X_data, self.slopes))
+
+            # Likelihood (sampling distribution) of observations
+            Y_obs = pm.Normal("y_obs", mu=self.mu, sigma=self.sigma, observed=y_data, dims =("obs_id"))
+
+            # Sample from the posterior
+            self.trace = pm.sample(
+                draws=2000,            
```

Comment:
should we make these available in the `__init__`? Can of course leave these to defaults.

Reply:
Yep; I've added the options to set these in `__init__`